MovieDirectory/MovieDirectory/Strategies/IActionMovie.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DataMovie;
+
+namespace MovieDirectory
+{ 
+    public interface IActionMovie
+    {
+        public void Action(Dictionary<int, Movie> movies, params object[] data);
```

Comment:
I suggest you using generic here, like `IActionMovie<T>`. This way you will be able to accept a specific type of `data` parameter.
For example, for create action you will be able to accept `CreateModel` and implement interface `IActionMovie<CreateModel>`.

Please read and learn how to work with generics, this is a really good example how to avoid using just `object` type